### PR TITLE
Widen "publish transaction" modal

### PR DESCRIPTION
**Pull request by brunoparga**
To make debugging easier (and also improve UX), I made the modal for transaction publishing wider, in particular the part where the user selects the income or invoice the txn refers to.

### CODE REVIEW

File: src/components/transactions/EditTransactionModal/AccountingLines.tsx [hunk 1]
```diff
@@ -54,16 +54,16 @@ export const AccountingLines = ({
                             className="relative top-3 right-0.5 w-7 h-8 flex items-center justify-center"
                             onClick={(): void => handleAddNewLine()}
                         >
-                            <CirclePlus className="w-6 h-6"/>
+                            <CirclePlus className="w-6 h-6" />
```
Comment (brunoparga) [inline src/components/transactions/EditTransactionModal/AccountingLines.tsx:57]: Just whitespace - this time it was Claude that did it, not VSCodium

File: src/components/transactions/EditTransactionModal/CollectionLine.tsx [hunk 2]
```diff
@@ -86,7 +86,7 @@ const CollectionLine = ({
       default:
         throw new Error('Illegal accountType option');
     }
-  }, [enableQuickbooksDataFetching, colIdx, accountingType, accounts, bills, invoices, items, currencyMappings, transaction.tokenIncoming, transaction.tokenOutgoing, token]);
+  }, [enableCurrencyMappingsFetching, enableQuickbooksDataFetching, colIdx, accountingType, accounts, bills, invoices, items, currencyMappings, transaction.tokenIncoming, transaction.tokenOutgoing, token]);
```
Comment (brunoparga) [inline src/components/transactions/EditTransactionModal/CollectionLine.tsx:89]: This dependency was missing

File: src/components/transactions/EditTransactionModal/index.tsx [hunk 3]
```diff
@@ -35,27 +35,57 @@ interface Props {
 
 const initializeAccountingLinesIncoming = (transaction: TransactionProps): AccountingLine[] => {
   if (transaction.direction === Direction.Swap) {
-    const accountingLinesSwapIn = transaction?.accountingLines?.filter(accountingLine => accountingLine.accountingType && AccountingTransactionType.Swap === accountingLine.accountingType) || [];
```
Comment (brunoparga) [inline src/components/transactions/EditTransactionModal/index.tsx:38]: I wanted a default AccountingLine to be present when the modal loads. In the process of adding it I streamlined the functions.

File: src/components/transactions/EditTransactionModal/index.tsx [hunk 4]
```diff
@@ -35,27 +35,57 @@ interface Props {
...
 };
 
 const EditTransactionModal = ({
   editTransaction,
   transaction,
   nonSetMappings,
 }: Props): React.JSX.Element => {
-  const [accountingLinesIncoming, setAccountingLinesIncoming] = useState<AccountingLine[]>(initializeAccountingLinesIncoming(transaction as TransactionProps));
-  const [accountingLinesOutgoing, setAccountingLinesOutgoing] = useState<AccountingLine[]>(initializeAccountingLinesOutgoing(transaction as TransactionProps));
+  const [accountingLinesIncoming, setAccountingLinesIncoming] = useState<AccountingLine[]>(
```
Comment (brunoparga) [inline src/components/transactions/EditTransactionModal/index.tsx:81]: Just whitespace

**Review by celadari — APPROVED** [review]

File: src/components/transactions/EditTransactionModal/index.tsx [hunk 5]
```diff
@@ -107,15 +139,37 @@ const EditTransactionModal = ({
...
         .map(tokenMapping => tokenMapping.token);
-    if (transaction?.direction === Direction.Incoming) return nonMappedTokens.filter(token => keyFormatTransaction(token) === keyFormatTransaction(transaction?.tokenIncoming as Token));
-    else if (transaction?.direction === Direction.Outgoing) return nonMappedTokens.filter(token => keyFormatTransaction(token) === keyFormatTransaction(transaction?.tokenOutgoing as Token));
-    else if (transaction?.direction === Direction.Swap) {
-      return nonMappedTokens.filter(token => keyFormatTransaction(token) === keyFormatTransaction(transaction?.tokenIncoming as Token) || keyFormatTransaction(token) === keyFormatTransaction(transaction?.tokenOutgoing as Token));
-    } else return [];
-  }, [nonSetMappings, transaction]);
+
+
+      switch (transaction?.direction) {
```
Comment (brunoparga) [inline src/components/transactions/EditTransactionModal/index.tsx:151]: The `switch` statement is more readable than the chain of `if...else if`.

**Pull request merged**